src/twisted/newsfragments/12388.removal

```diff
@@ -0,0 +1 @@
+Remove ``TestCase.deferSetUp``, ``TestCase.deferTestMethod``, ``TestCase.deferTearDown`` and ``TestCase.deferRunCleanups``.
```

Comment:
Tring to get this to our standard release notes format with the canonical name of the targeted API 

Also, add a note about why we did this .

```suggestion
twisted.trial.unittest.TestCase.deferSetUp, twisted.trial.unittest.TestCase.deferTestMethod, twisted.trial.unittest.TestCase.deferTearDown, and twisted.trial.unittest.TestCase.deferRunCleanups were removed and converted to private methods. These functions expose so much of the internal structure of TestCase that it makes the code hard to change.
```